### DISCUSSION TITLE
Fast Refresh takes 20+ seconds to trigger

### DISCUSSION THREAD

**Discussion by zbeyens** [vercel/next.js#59533] [App Router] [answered]
I'm having an issue regarding to Fast Refresh on all our pages. Currently, each change triggers Fast Refresh after at least 20 to 30 seconds.

Env: Mac M2, 32GB

**Comment by csvidit** [comment]
I think it might have something to do with the Next.js compiler built on SWC on v12+. I have been facing this issue since I was on Next.js 14. Which version are you on?

**Reply by zbeyens** [reply]
Next 14, but also happens in 13.5.6. It doesn't look to depend on the number of modules of the page (same perf between 500 and 5000 modules). Also, it looks like it's reactive as it should (refresh under a second) for a few seconds sometimes (but rarely), then it's gone again.

**Reply by zbeyens** [reply]
I can reproduce it in that repo: https://github.com/chhpt/nextjs-starter, so it looks like it's due to my device.

**Comment by zbeyens** [comment, marked as answer]
I got it fixed by rebooting my Mac. Uptime was 100 days 😅
